gui/src/ScenarioViewer.tsx

```diff
@@ -0,0 +1,608 @@
+        </IconButton>
+      </Grid>
+    </Grid>
+  );
+};
+
+const ScenarioViewer = (props: ScenarioViewerProps) => {
+  const {
+    id = "",
+    scenario = [
+      "test12",
+      true,
+      "test",
+      "test",
+      "test",
+      ["tags"],
+      [["key", "value"]],
+      [["key", "value"]],
+      ["test123"],
+    ],
+  } = props;
+
+  const [confirmDialogOpen, setConfirmDialogOpen] = useState(false);
+  const [properties, setProperties] = useState<Property[]>([]);
+  const [action, setAction] = useState<string>("EDIT");
+  const [tags, setTags] = useState<string[]>(scenario[5]);
+  const [newProp, setNewProp] = useState<Property>({
+    id: "",
+    key: "",
+    value: "",
+  });
+  const [label, setLabel] = useState<string>();
+
+  const sendScenario = useCallback((e: React.MouseEvent<HTMLElement>) => {},
+  []);
+
+  const updatePropertyField = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const { idx = "", name = "" } =
+        e.currentTarget.parentElement?.parentElement?.dataset || {};
+      if (name) {
+        if (idx) {
+          setProperties((props) =>
+            props.map((p, i) => {
+              if (idx == i + "") {
+                p[name as keyof Property] = e.target.value;
+              }
+              return p;
+            })
+          );
+        } else {
+          setNewProp((np) => ({ ...np, [name]: e.target.value }));
+        }
+      }
+    },
+    []
+  );
+
+  const propertyDelete = useCallback((e: React.MouseEvent<HTMLElement>) => {
+    const { id = "-1" } = e.currentTarget.dataset;
+    setProperties((props) => props.filter((item) => item.id !== id));
+  }, []);
+
+  const propertyAdd = () => {
+    setProperties((props) => [
+      ...props,
+      { ...newProp, id: props.length + 1 + "" },
+    ]);
+    setNewProp({ id: "", key: "", value: "" });
+  };
+
+  const onDeleteScenario = useCallback(() => {
+    setConfirmDialogOpen(false);
+    close();
+  }, [close]);
+
+  const onConfirmDialogOpen = useCallback(() => setConfirmDialogOpen(true), []);
+
+  const onConfirmDialogClose = useCallback(
+    () => setConfirmDialogOpen(false),
+    []
+  );
+
+  const MuiAccordionSummary = styled((props: AccordionSummaryProps) => (
+    <AccordionSummary
+      expandIcon={<ArrowForwardIosSharp sx={AccordionSummarySx} />}
+      {...props}
+    />
+  ))(({ theme }) => ({
+    flexDirection: "row-reverse",
+    "& .MuiAccordionSummary-expandIconWrapper.Mui-expanded": {
+      transform: "rotate(90deg)",
+      marginRight: theme.spacing(1),
+    },
+    "& .MuiAccordionSummary-content": {
+      marginLeft: theme.spacing(1),
+    },
+  }));
+
+  const [
+    scenarioId,
+    primary,
+    config,
+    date,
+    scenarioLabel,
+    _scenarios,
+    scenarioProperties,
+    pipelines,
+    scenarioTags,
+  ] = scenario;
+
+  return (
+    <div>
+      <Box sx={MainBoxSx}>
+        <Accordion>
+          <MuiAccordionSummary id={`panel-${id}`}>
```

Comment:
is this id necessary ?
If it is, don't use ${id} as id can be undefined and then the comp id would be panel-undefined which will not be unique
I guess we can use scenarioId if necessary